repeater/src/main/scala/com.snowplowanalytics.snowplow.storage.bigquery.repeater/Repeater.scala

```diff
@@ -43,36 +40,31 @@ object Repeater extends SafeIOApp {
             .PubSub
             .getEvents(
               resources.insertBlocker,
-              resources.env.config.projectId,
-              command.failedInsertsSub,
-              resources.desperates
+              resources.env.projectId,
+              resources.env.config.input.subscription,
+              resources.uninsertable
             )
             .interruptWhen(resources.stop)
             .through[IO, Unit](Flow.sink(resources))
-          desparatesSink = Flow.dequeueDesperates(resources)
-          logging        = Stream.awakeEvery[IO](5.minute).evalMap(_ => resources.updateLifetime *> resources.showStats)
-          _ <- Stream(bqSink, desparatesSink, logging).parJoin(
+          uninsertableSink = Flow.dequeueUninsertable(resources)
+          logging          = Stream.awakeEvery[IO](5.minute).evalMap(_ => resources.updateLifetime *> resources.showStats)
+          _ <- Stream(bqSink, uninsertableSink, logging).parJoin(
             StreamConcurrency
           )
         } yield ()
 
         process.compile.drain.attempt.flatMap {
           case Right(_) =>
             IO.delay(println("Closing Snowplow BigQuery Repeater")) *> IO.pure(ExitCode.Success)
-          case Left(Config.InitializationError(message)) =>
-            IO(
-              System
-                .err
-                .println(
-                  s"Snowplow BigQuery Repeater failed to start. $message"
-                )
-            ) >> IO.pure(ExitCode.Error)
           case Left(e: java.util.concurrent.TimeoutException) =>
             System.out.println(e.toString)
             IO.raiseError(e) >> IO.pure(ExitCode.Error)
           case Left(NonFatal(e)) =>
             System.out.println(e.toString)
             IO.raiseError(e) >> IO.pure(ExitCode.Error)
+          case Left(e) =>
```

Comment:
This smells wrong.  You have already handled `Left(NonFatal(e))` on a line above.  So this line would only match a **fatal** exception.  Generally we don't deliberately catch fatal exceptions - they are things like `VirtualMachineError` or `ThreadDeath`.  I haven't checked where this exception is coming from, but you will hopefully find that it is already guaranteed to be non fatal or else it would not have been caught and wrapped into an Either.